core/math/bvh_public.inc

```diff
@@ -328,6 +334,8 @@ void item_set_pairable(const BVHHandle &p_handle, bool p_pairable, uint32_t p_pa
 		// always keep this up to date
 		ex.pairable = p_pairable;
 	}
+
```

Comment:
I'm not sure the logic in this function appears exactly the same depending on the quick reject (although it may produce the same result, it is confusing to follow).

Actually given how little the cost of the full logic and the potential for bugs like this (and future bugs), I think it is worth running the full logic in both cases, the only relevant thing is whether the function returns true or false (and thus runs the collision check which is the expensive bit). That is unless we have evidence that this is a bottleneck from profiling.